Cpp/odin-views/link.account.cpp

```diff
@@ -0,0 +1,255 @@
+/**
+    Copyright 2018-2019 Red Anchor Trading Co. Ltd.
+
+    Distributed under the Boost Software License, Version 1.0.
+    See <http://www.boost.org/LICENSE_1_0.txt>
+ */
+
+
+#include <odin/app.hpp>
+#include <odin/odin.hpp>
+#include <odin/views.hpp>
+
+#include <fost/crypto>
+#include <fost/insert>
+#include <fost/json>
+#include <fost/log>
+#include <fostgres/sql.hpp>
+
+#include <pqxx/except>
+
+
+namespace {
+
+    inline std::pair<boost::shared_ptr<fostlib::mime>, int>
+            respond(fostlib::string message, int code = 403) {
+        fostlib::json ret;
+        if (not message.empty())
+            fostlib::insert(ret, "message", std::move(message));
+        fostlib::mime::mime_headers headers;
+        boost::shared_ptr<fostlib::mime> response(new fostlib::text_body(
+                fostlib::json::unparse(ret, true), headers, "application/json"));
+        return std::make_pair(response, code);
+    }
+
+    bool has_account_registerd(
+            fostlib::pg::connection &cnx, fostlib::string const identity_id) {
+        auto const credentials = cnx.exec(
+            "SELECT 1 FROM "
+            "odin.identity id "
+            "LEFT JOIN odin.credentials cred "
+            "ON cred.identity_id = id.id "
+            "LEFT JOIN odin.facebook_credentials fb "
+            "ON fb.identity_id = id.id "
+            "LEFT JOIN odin.google_credentials gg "
+            "ON gg.identity_id = id.id "
+            "WHERE id.id='" + identity_id + "' "
```

Comment:
Is this allow the SQL injection? @KayEss Any thought?

Reply:
Yes, that's exactly what that is. Let's not do that

Reply:
If not `cnx.exec`, what should we use?

Reply:
That's not the problem. Use `%s` in the SQL and then pass a tuple as an extra parameter. Look at the other examples.

Reply:
Can I use $1 as `WHERE id.id=$1` and
cnx.procedure(sql).exec(std::vector<fostlib::string>{identity_id})